lib/cloud_controller/diego/reporters/instances_stats_reporter.rb

```diff
@@ -92,15 +121,16 @@ def envelopes(desired_lrp, process)
           source_guid = process.guid
         end
 
-        @logstats_client.container_metrics(
+        return @logstats_client.container_metrics(
           source_guid: source_guid,
           auth_token: VCAP::CloudController::SecurityContext.auth_token,
           logcache_filter: filter
-        )
+        ), nil
+      rescue GRPC::BadStatus, CloudController::Errors::ApiError => e
+        logger.error('stats_for_app.error', error: e.message, backtrace: e.backtrace.join($INPUT_RECORD_SEPARATOR))
```

Comment:
what does this global variable do?

Reply:
It just joins them with the system's concept of what the newline separator should be. I think it's always `"\n"` in our case and it's probably better if I just explicitly declare that.

Reply:
Made the change:
https://github.com/cloudfoundry/cloud_controller_ng/pull/2250/files#diff-ad5dff07755b8d42aeadbf674ff7947d38ac428c9e7e52e415a6194d07ba202eR130